### PR TITLE
HPCC-29666 Fix resources not being picked up from dependent packages

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: ecl/eclcc/eclcc.cpp [hunk 1]
```diff
@@ -870,6 +867,24 @@ void gatherResourceManifestFilenames(EclCompileInstance & instance, StringArray
...
+
+}
+
+void gatherResourceManifestFilenames(EclCompileInstance & instance, StringArray &filenames)
+{
+    if (instance.archive)
+    {
+        gatherResourceManifestFilenames(instance.archive, "Module/Attribute", filenames);
+
+        //Gather resources from dependent packages
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:879]: This is the main bug fix for the jira - the new git dependencies have nested \<Archive>s with the main Archive

File: ecl/eclcc/eclcc.cpp [hunk 2]
```diff
@@ -2949,8 +2964,9 @@ int EclCC::parseCommandLineOptions(int argc, const char* argv[])
         else if (iter.matchFlag(optSaveQueryArchive, "-qa"))
         {
         }
-        else if (iter.matchOptionText(eclRepoPath, "--repocachepath", false, false))
+        else if (iter.matchOption(tempArg, "--repocachepath"))
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:2967]: This was invalid function to call - matchOptionText appends the option name as well for some reason.

File: ecl/eclccserver/eclccserver.cpp [hunk 3]
```diff
@@ -641,7 +641,7 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
         //Ensure that each child compile has a separate directory for the cloned git repositories.
         //It means the caches are not shared, but avoids the clones/fetches from affecting each other
         if (!repoRootPath.isEmpty())
-            eclccCmd.appendf(" --repocachepath=\"%s\"", repoRootPath.str());
+            eclccCmd.appendf(" \"--repocachepath=%s\"", repoRootPath.str());
```
Comment (ghalliday) [inline ecl/eclccserver/eclccserver.cpp:644]: Quote was in the wrong place - which meant the quotes were included in the filename

**Review by GordonSmith — APPROVED** [review]
Nothing stands out syntactically

**Pull request merged**